plugin/sampling/strategystore/static/strategy_store.go

```diff
@@ -136,6 +136,7 @@ func (h *strategyStore) parseStrategy(strategy *strategy) *sampling.SamplingStra
 		}
 	default:
 		h.logger.Warn("Failed to parse sampling strategy", zap.Any("strategy", strategy))
-		return &defaultStrategy
+		copyOfDefaultStrategy := defaultStrategy
+		return &copyOfDefaultStrategy
```

Comment:
I am not sure this way of copying is correct. It only works for primitive strategies, anything nested would still have the same issue. I think it needs a deep copy.